editor/store/reducer.js

```diff
@@ -115,7 +118,27 @@ export const editor = flow( [
 	combineReducers,
 
 	// Track undo history, starting at editor initialization.
-	partialRight( withHistory, { resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ] } ),
+	partialRight( withHistory, {
+		resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ],
+		shouldOverwriteState( action, previousAction ) {
```

Comment:
Wondering if we could avoid inlining this complex logic. Even just a top-level function. And document it.